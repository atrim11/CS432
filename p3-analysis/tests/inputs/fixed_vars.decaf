int x;
int y;
int z;
bool a;
bool b;
bool c;

def int something() {
    return x + y;
}

def int main() {
    return -1;
}

def void now() {}