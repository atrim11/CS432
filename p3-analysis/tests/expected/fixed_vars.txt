Program [line 1]
SYM TABLE:
 print_int : (int) -> void
 print_bool : (bool) -> void
 print_str : (str) -> void
 something : () -> int
 main : () -> int
 now : () -> void
 x : int
 y : int
 z : int
 a : bool
 b : bool
 c : bool

  FuncDecl name="something" return_type=int parameters={} [line 8]
  SYM TABLE:

    Block [line 8]
    SYM TABLE:

  FuncDecl name="main" return_type=int parameters={} [line 12]
  SYM TABLE:

    Block [line 12]
    SYM TABLE:

  FuncDecl name="now" return_type=void parameters={} [line 16]
  SYM TABLE:

    Block [line 16]
    SYM TABLE:

